### PR TITLE
fix(ckeditor): fixes basepath issue on some systems

### LINKED ISSUE #10724: Elgg 2.3.0: CKeditor broken (either issue in AMD module of CKeditor or Elgg core caching issue)
After denying in https://elgg.org/discussion/view/2594751/ckeditor-issue that CKeditor is broken I now also have a broken CKeditor on any browser with Elgg 2.3.0 and caching enabled.

Browser console gives me

> Failed to load resource: http://site.url/cache/1485283855/default/elgg/ckeditor/config.js
> the server responded with a status of 404 (Not Found)

and

> require.js:17 Uncaught Error: Script error for "elgg/ckeditor/config", needed by: elgg/ckeditor
> http://requirejs.org/docs/errors.html#scripterror
>     at makeError (require.js:17)
>     at HTMLScriptElement.onScriptError (require.js:117)

CKeditor instance loads fine with caching disabled or with at least no "cache" link in the root directory.

So, it might not be an issue of CKeditor but rather a bug with caching / cache rebuilding (maybe a problem connected or in addition of https://github.com/Elgg/Elgg/issues/10706)

### LINKED ISSUE #10758: Rename elgg/ckeditor.js to eliminate need for CKEDITOR_BASEPATH
CKeditor only gets confused because we load a file named `ckeditor.js` and it isn't what it expects. Renaming our wrapper to something like `elgg-ckeditor.js` would eliminate the need for worrying about this global constant being set early enough.

### PR DESCRIPTION
**Pull request by mrclay**
As the CKeditor library is not a real AMD library, its execution may not be controlled as well by RequireJS, causing a race condition with the module that sets the global `CKEDITOR_BASEPATH` constant.

This renames the module `elgg/ckeditor` to `elgg-ckeditor` which eliminates the need for setting `CKEDITOR_BASEPATH` at all. With the new filename, the CKeditor library is no longer confused when sniffing for its resources folder.

For BC, the now-unneeded modules have been left in place and deprecated, and the global `CKEDITOR_BASEPATH` is set via inline code in `elgg.js`.

Fixes #10724

- [ ] reword commit to fix #10758

### CODE REVIEW

**Comment by mrclay** [comment]
For BC, we really need to keep set-basepath defined in 2.3.

File: mod/ckeditor/views/default/elgg/ckeditor/set-basepath.js [hunk 1]
```diff
@@ -1,7 +1,13 @@
 /**
+ * This view exists for BC. Remove in Elgg 3.0.
+ *
  * This view extends elgg.js and sets ckeditor basepath before jquery.ckeditor is loaded
  */
 define('elgg/ckeditor/set-basepath', function (require) {
+	if (window && window.console) {
+		console.info("The 'elgg/ckeditor/set-basepath' module is unneeded and will be removed in Elgg 3.0");
```
Comment (hypeJunction) [inline mod/ckeditor/views/default/elgg/ckeditor/set-basepath.js]: This is going to be logged on each page view...

Comment (mrclay) [inline mod/ckeditor/views/default/elgg/ckeditor/set-basepath.js]: Only if a really customized plugin is requiring this, and even then... meh.

**Comment by mrclay** [comment]
@hypeJunction you cool with this for master?

**Comment by hypeJunction** [comment]
Sure, if it helps anybody's

**Comment by mrclay** [comment]
Replaced by #10874

**Pull request closed without merging**